### PR TITLE
ArangoExporter support

### LINKED ISSUE #257: Integrate arangodb-exporter
In order to export arangodb metrics to Prometheus, the operator should be extended to run an `arangodb-exporter` sidecar container for every `arangod`(agent|dbserver|coordinator|single) `Pod`.

This involves:

- [x] Adding an option to the `DeploymentSpec` to turn on/off exporting metrics
- [x] If enabled, add a container running `arangodb-exporter` for every `arangod` pod.
  Follow the template spec in the `Deployment` example below.
- [ ] Add a `Service` that selects all exporters of the deployment.
- [ ] Add a `ServiceMonitor` that adds the arangodb-exporters as target in Prometheus.

See https://github.com/arangodb-helper/arangodb-exporter/pull/4 for using the exporter with the Prometheus Operator.

Below is an example of (manually) running the exporter for 1 agent (that was deployed by the operator).

```yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: arangodb-export-agent1
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: arangodb-exporter
    spec:
      containers:
      - name: exporter
        image: arangodb/arangodb-exporter:0.1.3
        ports:
        - name: metrics
          containerPort: 9101
        args:
        - --arangodb.endpoint=ssl://example-simple-cluster-agent-0mmwt80z.example-simple-cluster-int.default.svc:8529
        - --arangodb.jwtsecret=$(ARANGOD_JWT_SECRET)
        - --ssl.keyfile=/secrets/tls/tls.keyfile
        env:
        - name: ARANGOD_JWT_SECRET
          valueFrom:
            secretKeyRef:
              key: token
              name: example-simple-cluster-jwt
        volumeMounts:
        - mountPath: /secrets/tls
          name: tls-keyfile
      volumes:
      - name: tls-keyfile
        secret:
          defaultMode: 420
          secretName: example-simple-cluster-agent-0mmwt80z-tls-keyfile

---

kind: Service
apiVersion: v1
metadata:
  name: arangodb-exporters-service
  labels:
    app: arangodb-exporter
spec:
  selector:
    app: arangodb-exporter
  ports:
  - name: metrics
    port: 9101

---

apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: arangodb-exporter
  namespace: monitoring
  labels:
    team: frontend
    prometheus: kube-prometheus
spec:
  namespaceSelector:
    matchNames:
    - default
  selector:
    matchLabels:
      app: arangodb-exporter
  endpoints:
  - port: metrics
    scheme: https
    tlsConfig:
      insecureSkipVerify: true
```

### PR DESCRIPTION
**Pull request by maierlars**
Add the exporter as side car for the arangod process. 

fixes #257

### CODE REVIEW

File: pkg/deployment/resources/pod_creator.go [hunk 1]
```diff
@@ -329,6 +329,23 @@ func createArangoSyncArgs(apiObject metav1.Object, spec api.DeploymentSpec, grou
 	return args
 }
 
+func createExporterArgs() []string {
+	options := make([]optionPair, 0, 64)
+	options = append(options,
+		optionPair{"--arangodb.jwtsecret", "$(" + constants.EnvArangodJWTSecret + ")"},
```
Comment (ewoutp) [inline pkg/deployment/resources/pod_creator.go]: Can we pass this by file? 
If not, let's extend the exporter to accept that first

File: pkg/util/k8sutil/pods.go [hunk 2]
```diff
@@ -329,6 +330,32 @@ func arangosyncContainer(image string, imagePullPolicy v1.PullPolicy, args []str
 	return c
 }
 
+func arangodbexporterContainer(image string, imagePullPolicy v1.PullPolicy, args []string, env map[string]EnvValue, livenessProbe *HTTPProbeConfig) v1.Container {
+	// THIS IS WORK IN PROGRESS
+	// IN THE END THE ARANGO DOCKERIMAGE WILL CONTAIN THE EXPORTER
+	c := v1.Container{
+		Command:         append([]string{"/app/arangodb-exporter"}, args...),
+		Name:            ExporterContainerName,
+		Image:           "arangodb/arangodb-exporter:0.1.3",
```
Comment (ewoutp) [inline pkg/util/k8sutil/pods.go]: Suggestion: Make the image configurable in the spec.
If not specified, use the standard arangodb image (as soon as it contains the exporter)

File: pkg/deployment/resources/pod_creator.go (hunk 1 above)
Comment (lamai93) [reply pkg/deployment/resources/pod_creator.go]: I would even prefer to not hand over the secret but a token. One could craft a specific token that allows access to statistics.

Comment (lamai93) [reply pkg/deployment/resources/pod_creator.go]: One drawback: a special token would require an extra secret when passed by file.

File: pkg/util/k8sutil/pods.go [hunk 3]
```diff
@@ -94,13 +97,42 @@ func IsPodReady(pod *v1.Pod) bool {
...
+			if t != nil {
+				return t.ExitCode == 0
+			}
+		}
+		return false
+	}
 }
 
 // IsPodFailed returns true if all containers of the pod
 // have terminated and at least one of them wih a non-zero exit code.
```
Comment (neunhoef) [inline pkg/util/k8sutil/pods.go]: Please adjust comment.

**Pull request merged**